server/shared/types/dataset.ts

```diff
@@ -478,31 +477,57 @@ type GDCSingleCellPlot = {
 	colorColumn: ColorColumn
 	coordsColumns: { x: number; y: number }
 }
+
 type ColorColumn = {
+	/** 0-based column number in the tabular file */
 	index?: number
+	/** name of the column */
 	name: string
+	/** column values (categories) to color mapping */
 	colorMap?: { [index: string]: string }
 }
+
+/** defines a tsne type of plot for cells from one sample */
 type SingleCellPlot = {
+	/** type of the plot, e.g. tsne or umap, also display as plot name on ui */
 	name: string
+	/** folder in which per-sample files are stored.
+	each file is a tabular text file with all cells (rows) from that sample.
+	all files must have same set of columns
+	file columns include cell types and x/y coords, as described by other parameters
+	*/
 	folder: string
-	fileSuffix: string
+	/** optional suffix to locate the file for a sample, via ${folder}/${sampleName}${fileSuffix}
+	assumes that file name always start with sample name.
+	if not introduce filePrefix
+	*/
+	fileSuffix?: string
+	/** specify which column to color the cells in the plot. must have categorical values
+	TODO define a set of color columns and specify a default one, and let ui to switch from
+	*/
 	colorColumn: ColorColumn
+	/** 0-based column number for x/y coordinate for this plot */
 	coordsColumns: { x: number; y: number }
 }
 export type SingleCellDataNative = {
-	src: 'native' | string
+	src: 'native'
+	/** when a sample has multiple tsne plots, this flag allows allows all plots to share one cell type legend */
 	sameLegend: boolean
+	/** available tsne type of plots for each sample */
 	plots: SingleCellPlot[]
-	refName: string
+	/** name of ref cells? */
+	refName?: string
```

Comment:
Airen please clarify the purpose of this

Reply:
This is the code behind 
`itemG
				.append('g')
				.attr('transform', `translate(${x + 10}, ${5})`)
				.append('text')
				.text(
					`${
						cluster == 'ref'
							? this.state.termdbConfig.queries.singleCell.data.refName
							: cluster == 'query'
							? this.state.config.sample
							: cluster
					} n=${n}`
				)`